packages/deployed-backend-client/src/stack_metadata_output_retrieval_strategy.ts

```diff
@@ -37,18 +37,37 @@ export class StackMetadataBackendOutputRetrievalStrategy
   fetchBackendOutput = async (): Promise<BackendOutput> => {
     const stackName = await this.stackNameResolver.resolveMainStackName();
 
-    // GetTemplateSummary includes the template metadata as a string
-    const templateSummary = await this.cfnClient.send(
-      new GetTemplateSummaryCommand({ StackName: stackName })
-    );
-    if (typeof templateSummary.Metadata !== 'string') {
-      throw new BackendOutputClientError(
-        BackendOutputClientErrorType.METADATA_RETRIEVAL_ERROR,
-        'Stack template metadata is not a string'
+    let metadataObject;
+
+    try {
+      // GetTemplateSummary includes the template metadata as a string
+      const templateSummary = await this.cfnClient.send(
+        new GetTemplateSummaryCommand({ StackName: stackName })
       );
-    }
 
-    const metadataObject = JSON.parse(templateSummary.Metadata);
+      if (typeof templateSummary.Metadata !== 'string') {
+        throw new BackendOutputClientError(
+          BackendOutputClientErrorType.METADATA_RETRIEVAL_ERROR,
+          'Stack template metadata is not a string'
+        );
+      }
+
+      metadataObject = JSON.parse(templateSummary.Metadata);
+    } catch (error) {
+      if (
+        error instanceof Error &&
+        error.name === BackendOutputClientErrorType.VALIDATION_ERROR &&
+        error.message.startsWith('Stack with id') &&
+        error.message.endsWith('does not exist')
```

Comment:
I'm guessing the error is thrown by CFN service? If yes, can we switch validation to be against the CFN exception types/name

Reply:
Done in https://github.com/aws-amplify/amplify-backend/pull/1790/commits/42d18fb3745cf52c777b026ea7d7281aa15c42ae